optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/locator/LocationStrategyResolver.java

```diff
@@ -88,6 +89,21 @@ public LocationStrategy determineLocationStrategy(Object object) {
                     }
                     return new PlanningIdLocationStrategy(memberAccessor2);
                 case EQUALITY:
+                    try {
+                        Method equals = object.getClass().getMethod("equals", Object.class);
+                        if (equals.getDeclaringClass().equals(Object.class)) {
+                            throw new IllegalArgumentException("The class (" + object.getClass().getSimpleName()
+                                    + ") doesn't override equals method, neither does any superclass.");
+                        }
+                        Method hashCode = object.getClass().getMethod("hashCode");
+                        if (hashCode.getDeclaringClass().equals(Object.class)) {
+                            throw new IllegalArgumentException("The class (" + object.getClass().getSimpleName()
+                                    + ") overrides equals but neither it nor any superclass overrides hashCode method."
+                            );
+                        }
```

Comment:
Good question, but I think there might be valid cases indeed for which they are not overridden in the same class, as the example you gave. We can't afford any "false negatives", but we can afford a little as possible "false positives". So I 'd leave it as is. It might not fail fast on all negatives, but it never fails fast when it shouldn't.